internal/utils/reader_test.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"bytes"
+	"io"
+	"testing"
+)
+
+func TestReaderAtAdapter_Read(t *testing.T) {
+	data := []byte("Hello, World!")
+	buffer := bytes.NewReader(data)
+	reader := NewReaderAtAdapter(buffer)
+
+	// Test reading the entire data
+	readData := make([]byte, len(data))
+	n, err := reader.Read(readData)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
```

Comment:
`Errorf` also set failed status, but did not stop execution immediately. I don't think it matters because in most cases this function should not be called. The test should pass correctly.